src/config.cpp

```diff
@@ -103,99 +189,170 @@ const Config& config()
         ("clockid,k",
-             po::value(&requested_clock_name)->default_value("monotonic-raw"),
-             "clock used for perf timestamps (see --list-clockids for supported arguments)")
-#ifdef HAVE_X86_ADAPT // I am going to burn in hell for this
-        ("x86-adapt-cpu-knob,x", po::value(&config.x86_adapt_cpu_knobs),
-             "add x86_adapt knobs as recordings. Append #accumulated_last for semantics.")
-#endif
-        ("disassemble", po::bool_switch(&disassemble),
-             "enable augmentation of samples with instructions (default if supported)")
-        ("no-disassemble", po::bool_switch(&no_disassemble),
-             "disable augmentation of samples with instructions")
-        ("kernel", po::bool_switch(&kernel),
-             "include events happening in kernel space (default)")
-        ("no-kernel", po::bool_switch(&no_kernel),
-             "exclude events happening in kernel space")
-        ("list-clockids", po::bool_switch(&list_clockids)->default_value(false),
-            "list all available clockids")
-        ("list-events", po::bool_switch(&list_events)->default_value(false),
-            "list all available events")
-        ("command", po::value(&config.command));
+            po::value(&requested_clock_name)
+                ->value_name("CLOCKID")
+                ->default_value("monotonic-raw"),
+            "Clock used for perf timestamps (see --list-clockids for supported arguments).")
+        ("list-clockids",
+            po::bool_switch(&list_clockids)
+                ->default_value(false),
+            "List all available clockids.")
+        ("list-events",
+            po::value(&list_event_category)
+                ->value_name("CATEGORY")
+                ->implicit_value("all"),
+            "List available metric and sampling events from a given category.");
+
+    system_wide_options.add_options()
+        ("all-cpus,a",
+            po::bool_switch(&all_cpus),
+            "System-wide monitoring of all CPUs.");
+
+    sampling_options.add_options()
+        ("command",
+            po::value(&config.command))
+        ("count,c",
+            po::value(&config.sampling_period)
+                ->value_name("N")
+                ->default_value(11010113),
+            "Sampling period (in number of events specified by -e).")
+        ("event,e",
+            po::value(&config.sampling_event)
+                ->value_name("EVENT")
+                ->default_value("instructions"),
+            "Interrupt source event for sampling.")
+        ("call-graph,g",
+            po::bool_switch(&config.enable_cct),
+            "Enable call-graph recording.")
+        ("no-ip,n",
+            po::bool_switch(&config.suppress_ip),
+            "Do not record instruction pointers [NOT CURRENTLY SUPPORTED]")
+        ("pid,p",
+            po::value(&config.pid)
+                ->value_name("PID")
+                ->default_value(-1),
+            "Attach to process of given PID.")
```

Comment:
Can we hide the `(=-1)` in the output? Probably something like remove the `default_value()` and add something like this: 

```
if vm.count("pid"):
  config.pid = vm.pid
else:
  config.pid = -1
```